lib/src/Base/Stat/IsotropicCovarianceModel.cxx

```diff
@@ -0,0 +1,270 @@
+  const Point tau = t - s;
+  const Scalar tauNorm = tau.norm();
+  const Point normGradient(tau / tauNorm);
+  const Matrix kernelPartialGradient = kernel_.partialGradient(Point(1, tauNorm), Point(1));
+  return Matrix(inputDimension_, 1, normGradient) * kernelPartialGradient(0, 0);
+}
+
+void IsotropicCovarianceModel::setFullParameter(const Point & parameter)
+{
+  kernel_.setFullParameter(parameter);
+}
+
+Point IsotropicCovarianceModel::getFullParameter() const
+{
+  return kernel_.getFullParameter();
+}
+
+Description IsotropicCovarianceModel::getFullParameterDescription() const
+{
+  return kernel_.getFullParameterDescription();
+}
+
+/** Scale accessor */
+void IsotropicCovarianceModel::setScale(const Point & scale)
+{
+  if (scale.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the scale should have dimension 1, not " << scale.getDimension() << ".";
+  kernel_.setScale(scale);
+  scale_ = scale;
+}
+
+/* Nugget factor accessor */
+void IsotropicCovarianceModel::setNuggetFactor(const Scalar nuggetFactor)
+{
+  kernel_.setNuggetFactor(nuggetFactor);
+  nuggetFactor_ = nuggetFactor;
+}
+
+/** Indices of the active parameters */
+void IsotropicCovarianceModel::setActiveParameter(const Indices & active)
+{
+  kernel_.setActiveParameter(active);
+  activeParameter_ = active;
+}
+
+/** Amplitude accessor */
+void IsotropicCovarianceModel::setAmplitude(const Point & amplitude)
+{
+  if (amplitude.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the amplitude should have dimension 1, not " << amplitude.getDimension() << ".";
+  kernel_.setAmplitude(amplitude);
+  amplitude_ = amplitude;
+  updateOutputCovariance();
+}
+
+/* Output correlation accessor */
+void IsotropicCovarianceModel::setOutputCorrelation(const CorrelationMatrix & outputCorrelation)
+{
+  if (outputCorrelation.getDimension() != getOutputDimension())
+    throw InvalidArgumentException(HERE) << "The given spatial correlation should have dimension 1, not " << outputCorrelation.getDimension() << ".";
+  kernel_.setOutputCorrelation(outputCorrelation);
+  outputCorrelation_ = outputCorrelation;
+  updateOutputCovariance();
+}
+
+/* String converter */
+String IsotropicCovarianceModel::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << IsotropicCovarianceModel::GetClassName()
+      << " input dimension=" << inputDimension_
+      << ", dimension = " << getOutputDimension()
+      << ", kernel =" << kernel_;
+  return oss;
+}
+
+/* Is it a stationary model ? */
+Bool IsotropicCovarianceModel::isStationary() const
+{
+  return true;
+}
+
+/* String converter */
+String IsotropicCovarianceModel::__str__(const String & ) const
+{
+  return __repr__();
+}
+
+/* Update output covariance */
+void IsotropicCovarianceModel::updateOutputCovariance()
+{
+  //kernel_.updateOutputCovariance(); // does not work since the method is protected
+  // We can do this indirectly.
+  // kernel_.setOutputCorrelation calls kernel_.updateOutputCovariance.
```

Comment:
Ugly. Find something else.

Reply:
Hint: maybe a friend declaration is the option here

Reply:
Actually, I now think that doing nothing is a valid option. After all, the 1D kernel can take care of its own `outputCovariance_` and `outputCovarianceCholeskyFactor_` attributes  as long as its `outputCorrelation_` and `amplitude_` attributes are correctly set. So `IsotropicCovarianceModel::updateOutputCovariance`only needs to update the `outputCovariance_` and `outputCovarianceCholeskyFactor_` attributes of the `IsotropicCovarianceModel` object.